File: src/Optional.scala
case class User(
  firstName: String,
  optionalSecondName: Option[String],
  lastName: String
)


object NameLengthComputer {
  def fullNameLength(user: User) {
    val separatorsLength = 2
    user.firstName.length + user.optionalSecondName.length + user.lastName.length + separatorsLength
  }
}

File: src/Optional.java
class User {
  private String firstName;
  private String secondName;
  private String lastName;

  User(String firstName, String secondName, String lastName) {
    this.firstName = firstName;
    this.secondName = secondName;
    this.lastName = lastName;
  }

  public String getFirstName() { return firstName; }
  public String getSecondName() { return secondName; }
  public String getLastName() { return lastName; }
}

class NameLengthComputer {
  static int fullNameLength(User user) {
    int separatorsLength = 2;
    return user.getFirstName().length() + user.getSecondName().length() + user.getLastName().length() + separatorsLength;
  }
}
